ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -418,6 +419,27 @@ func (r *RayJobReconciler) releaseComputeResources(ctx context.Context, rayJobIn
 			r.Recorder.Eventf(rayJobInstance, corev1.EventTypeNormal, "Deleted", "Deleted cluster %s", rayJobInstance.Status.RayClusterName)
 		}
 	}
+
+	// Since the name of the Kubernetes Job is the same as the RayJob, we need to set the TTL of the Kubernetes Job to clean
+	// up the Kubernetes Job and its Pods when suspending, and a new submitter Kubernetes Job must be created to resubmit the
+	// Ray job if the RayJob is resumed.
+	if isSuspend {
+		jobName := rayJobInstance.Name
+		job := &batchv1.Job{}
+		if err := r.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: jobName}, job); err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("The submitter Kubernetes Job has been already deleted", "RayJob", rayJobInstance.Name, "Kubernetes Job", job.Name)
+			} else {
+				r.Log.Error(err, "Failed to get Kubernetes Job")
+				return err
+			}
+		}
+		job.Spec.TTLSecondsAfterFinished = pointer.Int32(0)
```

Comment:
I have currently set this to zero to avoid some race conditions. We can set a non-zero value after the `suspend` operation becomes irreversible. Additionally, the original `suspend` support (0.5.0?) also does not include a grace period.